Sources/SwiftSyntax/Syntax.swift

```diff
@@ -299,6 +304,11 @@ extension Syntax {
     return base.isExpr
   }
 
+  /// Whether or not this node represents an SyntaxCollection.
+  public var isSyntaxCollection: Bool {
```

Comment:
I like this, but minor nitpick: maybe the public-facing API name is a bit better as `isCollection` ? Having `Syntax` in `isSyntaxCollection` seems a bit redundant. For example it is `isDecl`, not `isDeclSyntax`.

Reply:
Done